src/tests/relay/t_relay_io_silo.cpp

```diff
@@ -1781,12 +1786,60 @@ TEST(conduit_relay_io_silo, read_silo)
         {
             remove_path_if_exists(out_name + "_write_overlink");
             write_opts["file_style"] = "overlink";
-            write_opts["ovl_topo_name"] = meshname;
+            write_opts["ovl_topo_name"] = meshname; // TODO do I even need this
             io::silo::save_mesh(load_mesh, out_name + "_write_overlink", write_opts);
         }
     }
 }
 
+//-----------------------------------------------------------------------------
+// test that we can read silo without multimeshes, multivars, and multimats
+TEST(conduit_relay_io_silo, read_simple_silo)
+{
+    const std::vector<std::vector<std::string>> file_info = {
+        {"curv2d",          ".silo", "no"},
+        {"curv2d_colmajor", ".silo", "no"},
+        {"curv3d",          ".silo", "yes"},
+        {"curv3d_colmajor", ".silo", "no"},
+        // {"globe",           ".silo", "yes"}, // TODO need to add support for mixed shape topos
+    };
+    for (int i = 0; i < file_info.size(); i ++) 
+    {
+        const std::string basename   = file_info[i][0];
+        const std::string fileext    = file_info[i][1];
+        const std::string round_trip = file_info[i][2];
+
+        Node load_mesh, info, write_opts;
+        std::string filepath = basename + fileext;
+        filepath = utils::join_file_path("silo", filepath);
+        std::string input_file = relay_test_silo_data_path(filepath);
+
+        io::silo::load_mesh(input_file, load_mesh);
+        EXPECT_TRUE(blueprint::mesh::verify(load_mesh, info));
+
+        const std::string out_name = "read_silo_" + basename;
+
+        // TODO are these remove paths doing anything? Don't they need filenames?
+        remove_path_if_exists(out_name + "_write_blueprint");
+        io::blueprint::save_mesh(load_mesh, out_name + "_write_blueprint", "hdf5");
+
+        if (round_trip == "yes")
+        {
+            remove_path_if_exists(out_name + "_write_silo");
+            io::silo::save_mesh(load_mesh, out_name + "_write_silo");
+
+            // overlink requires matsets
+            if (load_mesh[0].has_child("matsets"))
+            {
+                remove_path_if_exists(out_name + "_write_overlink");
+                write_opts["file_style"] = "overlink";
+                write_opts["ovl_topo_name"] = "MMESH"; // TODO do I even need this
```

Comment:
Does overlink ignore this (it always has to be MMESH?)

Reply:
MMESH is actually the name of the blueprint topo in this case, since we read an overlink file into blueprint.